syntax = "proto3";

package processing;

service Geocoder {
	rpc GeocodeCSV(GeocodeCSVRequest)
		returns (GeocodeCSVResponse) {}

	rpc PullProcessedData(PullProcessedDataRequest)
		returns (PullProcessedDataResponse) {}
}

message GeocodableData {
	bytes csv = 1;
	string address_column = 2;
	string tag = 3;
	string output_table = 4;
}

message GeocodeCSVRequest {
	GeocodableData data = 1;
}

message StoredRequest {
	enum RequestStatus {
		UNKNOWN = 0;
		RECIEVED = 1;
		REQUESTED = 2;
		COMPLETED = 3;
	}

	GeocodableData data = 1;
	RequestStatus status = 2;
	int64 group_tag = 3;
	string output_table = 4;
}

message GeocodeCSVResponse {
	int32 error = 1;
	string comment = 2;
	int32 unique_id = 3;
}

message PullProcessedDataRequest {
	bool delete_pulled_data = 1;
	string filter_by_tag = 2;
}

message PullProcessedDataResponse {
	repeated GeocodableData data = 1;
}
